File: MapboxTest/MapboxTest/ViewController.m
//
//  ViewController.m
//  MapboxTest
//
//  Created by zhangpan on 15/12/4.
//  Copyright © 2015年 wanttogo. All rights reserved.
//

#import "ViewController.h"
#import "Mapbox.h"

@interface ViewController ()

@property (nonatomic) MGLMapView *mapView;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    self.mapView = [[MGLMapView alloc] initWithFrame:self.view.bounds];
    self.mapView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
    self.mapView.styleURL = [MGLStyle lightStyleURL];
    
    // set the map's center coordinate
    [self.mapView setCenterCoordinate:CLLocationCoordinate2DMake(38.894368, -77.036487)
                            zoomLevel:15
                             animated:NO];
    [self.view addSubview:self.mapView];
    
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: InitializerTest/InitializerTest/ViewController.swift
//
//  ViewController.swift
//  InitializerTest
//
//  Created by zhangpan on 15/11/4.
//  Copyright © 2015年 wanttogo. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
//        let b = BBB(aaa: "")
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

class AAA {
    
    init(a: String) {
        
    }
    
    convenience init(b: String) {
//        self.init(a: b)
//        self.init(cc: b)!
        self.init(dd: b)!
    }
    
    required init(aa: String) {
        
    }
    
    init?(cc: String) {
        
    }
    
    convenience init?(dd: String) {
//        self.init(aaa: dd)
//        self.init(a: dd)
//        self.init(aa: dd)
        self.init(cc: dd)
    }
    
    required init?(aaa: String) {
        
    }
    
    required convenience init?(bbb: String) {
        self.init(cc: bbb)
    }
    
    
}

class BBB: AAA {
    required init(aa: String) {
        super.init(aa: aa)
    }
    
    required convenience init?(aaa: String) {
        self.init(aa: aaa)
    }
    
    required convenience init?(bbb: String) {
        self.init(aaa: bbb)
    }
    
    override init(cc: String) {
        super.init(aaa: cc)!
    }
    
    
}


